designer/server/src/main/scala/db/migration/V1_041__MoveTypePropertiesToGenericDefinition.scala

```diff
@@ -0,0 +1,22 @@
+package db.migration
+
+import db.migration.V1_041__MoveTypePropertiesToGenericDefinition.migrateMetaData
+import io.circe._
+import pl.touk.nussknacker.ui.db.migration.ProcessJsonMigration
+import io.circe.syntax._
+import pl.touk.nussknacker.engine.canonicalgraph.CanonicalProcess
+
+trait V1_041__MoveTypePropertiesToGenericDefinition extends ProcessJsonMigration {
+
+  override def updateProcessJson(json: Json): Option[Json] = migrateMetaData(json)
+
+}
+
+object V1_041__MoveTypePropertiesToGenericDefinition {
+
+  def migrateMetaData(json: Json): Option[Json] = json.as[CanonicalProcess] match {
+    case Left(failed: DecodingFailure) => throw new IllegalStateException(s"Migration failed - invalid json. $failed")
```

Comment:
The approach is ok, but I'm not sure if this error ever be thrown. I see that you have some decodeOption getOrElse in `MetaData`. It looks like you are eating errors. Doesn't it? Also, some comment will be nice if you have some not obvious assumptions.

Reply:
The intention was that by throwing the exception this would rollback the unsuccessful migration. The decoder should work in every case, however I wanted to leave this as an additional safety mechanism. I will check this further and maybe remove it.